doc/lsst.ap.pipe/apdb.rst

```diff
@@ -28,14 +28,13 @@ Configuring the database
 ========================
 
 The database is configured using `~lsst.dax.apdb.ApdbConfig`.
-|ap_pipe| command line users can pass configuration information to the script through the :option:`--config <ap_pipe.py --config>` and :option:`--configfile <ap_pipe.py --configfile>` command-line options.
-|make_apdb| also uses `ApPipeConfig` and the :option:`--config` and :option:`--configfile` options, so users can pass exactly the same arguments to |make_apdb| and |ap_pipe|.
-Supporting identical command line arguments for both scripts makes it easy to keep the database settings in sync.
+|ap_pipe| command line users can pass configuration information to the script through the :option:`--config <ap_pipe.py --config>` and :option:`--configfile <ap_pipe.py --configfile>` command-line options, using the prefix ``diaPipe.apdb.`` to distinguish APDB information from other pipeline configuration.
```

Comment:
I think it would be pretty easy to add `--config-file` as an alias to `CmdLineTask`, and I _think_ we can avoid an RFC on the "nobody would seriously object" grounds. Though I might want to ping people on `#dm` just to avoid accusations of messing with shared code...

The `.` vs `:` distinction is more fundamental to both the Gen 2 and Gen 3 parsers, and I think we just have to live with it. Sorry!

I think both errors are fairly fail-fast, though. If you switch `--config-file` and `--configfile` you'll get a "No such option" error. If you don't use `:` in Gen 3 get "unrecognized option syntax". Using `:` in Gen 2 would probably get you a "config does not have a `diaPipe:apdb` field". Hopefully that's enough if we warn users about the differences.

Reply:
Fair, I'm OK overlooking this concern here since it's a bit bigger than this ticket. I'd love a lower-level equivalent to what you implemented for `diaPipe.apdb.stuff` returning "It looks like you typed `diaPipe.apdb.stuff`, maybe try `stuff`" for the `configfile` vs. `config-file` situation. I don't quite know how to write that ticket, but sounds like you've started the conversation on #dm-middleware - thanks.